ToDo/src/components/Task/index.tsx

```diff
@@ -0,0 +1,202 @@
+import React, {useState} from 'react';
+import {View, Text, Pressable, StyleSheet} from 'react-native';
+import {ITask} from '../../types/common';
+import Icon from 'react-native-vector-icons/Feather';
+import {useAppDispatch} from '../../hooks/hooks';
+import {
+  completeTask,
+  deleteTask,
+  updateTask,
+} from '../../store/slices/tasksSlice';
+import TaskForm from '../TaskForm';
+import {
+  PanGestureHandler,
+  PanGestureHandlerGestureEvent,
+} from 'react-native-gesture-handler';
+import Animated, {
+  runOnJS,
+  useAnimatedGestureHandler,
+  useAnimatedStyle,
+  useSharedValue,
+  withTiming,
+} from 'react-native-reanimated';
+import {Dimensions} from 'react-native';
+
+interface TaskProps {
+  task: ITask;
+}
+
+const windowWidth = Dimensions.get('window').width;
```

Comment:
This API doesn't work well when rotating the device. The component will change orientation, re-render but the `windowWidth` will remain the same. Try getting this width inside component or use `useWindowDimensions` hook